src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -450,6 +469,32 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return cell;
 	}
 
+	/** Adds Paramters cell based on Notebook URI parameters */
+	private addParametersCell(notebookUriParams: string, hasParameterCell: boolean, parameterCellIndex: number, hasInjectedCell: boolean): void {
+		let uriParamsIndex = parameterCellIndex;
+		// Set new uri parameters as a Injected Parameters cell after original parameter cell
+		if (hasParameterCell) {
+			uriParamsIndex = parameterCellIndex + 1;
+			// Set the uri parameters after the injected parameter cell
+			if (hasInjectedCell) {
+				uriParamsIndex = uriParamsIndex + 1;
+			}
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isInjectedParameter = true;
+			this.cells[uriParamsIndex].source = [injectedParametersMsg].concat(this.cells[uriParamsIndex].source);
+		} else {
+			// Set new parameters as the parameters cell as the first cell in the notebook
+			this.addUriParameterCell(uriParamsIndex, notebookUriParams);
+			this.cells[uriParamsIndex].isParameter = true;
```

Comment:
So if there's already a parameters cell then we create a new cell that's an "injected parameters" cell. But if we don't then we create a normal parameters cell? Why the difference? I would expect that we either always add the injected parameters into a parameters cell or we create a new injected parameters cell each time.

Reply:
This is currently the designed plan from what has been discussed. 
The reason for this is that if a user goes from a normal notebook with absolutely zero parameters, and would like to turn this notebook (hosted somewhere) into a parameterized notebook (properly) then we should create a normal parameters cell (with the visual tag). As a result, the user can use this now as their "original" parameterized notebook. They can then host this, and continue chaining of parameterized notebooks (this work will enable this future work) with always having the original parameters notebook normal parameter cell noted.